src/bp-core/bp-core-catchuri.php

```diff
@@ -1543,3 +1543,93 @@ function bp_core_change_privacy_policy_link_on_private_network( $link, $privacy_
 
 	return $link;
 }
+
+/**
+ * Function will remove default wordpress and woocommerce api endpoint.
+ * Also, remove feeds url.
+ */
+add_action('init', 'bb_removes_endpoints_and_feeds');
+function bb_removes_endpoints_and_feeds() {
+	if ( ! is_user_logged_in() ) {
+		$enable_private_network    = bp_enable_private_network();
+		$buddyboss_app_plugin_file = 'buddyboss-app/buddyboss-app.php';
+		// IF Platform active and App plugin not active.
+		if ( defined( 'BP_PLATFORM_VERSION' ) && ! is_plugin_active( $buddyboss_app_plugin_file ) ) {
+			// IF Platform 'Private Website' settings enabled.
+			// THEN Restrict all RSS and REST APIs. (reason to restrict because the site is private).
+			if ( ! $enable_private_network ) {
+				bb_restricate_rss_feed();
+				bb_restricate_rest_api();
+			}
+		}
+		// IF Platform active and App plugin also active.
+		if ( defined( 'BP_PLATFORM_VERSION' ) && is_plugin_active( $buddyboss_app_plugin_file ) ) {
+			// IF Platform 'Private Website' settings enabled and App plugin 'Private App' settings disabled.
+			// THEN Restrict all RSS but do not restrict REST APIs.
+			// (reason to restrict RSS because the site is private)
+			// (reason to not restrict rest API because a customer might want to restrict site on the web but not on App)
+			if ( ! $enable_private_network && ! bbapp_is_private_app_enabled() ) {
+				bb_restricate_rss_feed();
+			}
+			// IF Platform 'Private Website' settings enabled and App plugin 'Private App' settings also enabled.
+			// THEN Restrict all RSS and REST APIs.
+			// (reason to restrict because the site is private on both web and app)
+			if ( ! $enable_private_network && bbapp_is_private_app_enabled() ) {
+				bb_restricate_rss_feed();
+				bb_restricate_rest_api();
+			}
+		}
+	}
+}
```

Comment:
@jitendrabanjara1991 can we simply update this function like this?

```suggestion
function bb_removes_endpoints_and_feeds() {
	if ( is_user_logged_in() ) {
		return;
	}

	if ( ! bp_enable_private_network() ) {
		bb_restricate_rss_feed();
	}

	if ( bp_rest_enable_private_network() ) {
		bb_restricate_rest_api();
	}
}
```

Reply:
@KartikSuthar 
Yes this will work. I have checked with and update it.
I have attached all condition video here.
1) platform active + app deactive
	Enable private website on platform
	Result - Restrict all RSS and REST APIs.
	working - https://screencast-o-matic.com/watch/crhX2DVfbtV
	
2) platform active + app active
	Enable private website on platform
	Disable private app on app
	Result - Restrict all RSS but do not restrict REST APIs.
	working - https://screencast-o-matic.com/watch/crhX2FVfbta

3) platform active + app active
	Enable private website on platform
	Enable private app on app
	Result - Restrict all RSS and REST APIs.
	working - https://screencast-o-matic.com/watch/crhX2YVfbue

Please check it and let me know if anything else.
Thanks